sources/abstract-parent/pom.xml

```diff
@@ -0,0 +1,44 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<project xmlns="http://maven.apache.org/POM/4.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
+    <modelVersion>4.0.0</modelVersion>
+    <parent>
+        <artifactId>smallrye-config-parent</artifactId>
+        <groupId>io.smallrye</groupId>
+        <version>1.3.12-SNAPSHOT</version>
+        <relativePath>../../pom.xml</relativePath>
+    </parent>
+
+    <groupId>io.smallrye.config</groupId>
+    <artifactId>smallrye-config-source-abstract-parent</artifactId>
+    <name>SmallRye: Abstract ConfigSource</name>
```

Comment:
Is this maybe a little overzealous?  It makes sense IMO to keep the base classes in the root project, lest the number of artifacts get truly out of hand.

Reply:
How would the base class be used by the extracted config sources, but also have those config sources as dependencies to the main implementation?

Reply:
On the issue I suggested using service loader on the implementation's own class loader, with a service type of e.g. `interface DefaultConfigurationSourceProvider extends ConfigurationSourceProvider` to load up the default config source instances (during static init of `SmallRyeConfigBuilder` perhaps).  The broken out config sources would depend on the main implementation then (rather than the reverse), fitting the SPI/provider pattern pretty well.

Also I think we do want the separated config source impls to be able to take advantage of any API extensions offered by the base implementation, so that would be another reason to do it that way.

Reply:
Sorry, I missed that comment on the issue. Will look to do that

Reply:
One thing with this approach, that means anything using SRye Config would need to include the impl and all the other config source artifacts, correct?

Should we rename the current impl to `smallrye-config-core` and then have a `smallrye-config` that bundles the default config sources as dependencies to simplify usage for downstream?

Reply:
Maybe we should keep the couple of spec-mandated default classes in the core and only break out the extra ones?

Otherwise having a dependency-aggregating artifact seems reasonable - I don't think we'd want to make a shaded aggregate though.  That could cause conflicts/confusion.

Reply:
Agree on no shading, that's nasty.

So do we aggregate or just leave in core?

Reply:
It's a small number of classes; let's leave it in core.